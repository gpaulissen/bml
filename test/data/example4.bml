(1N)---
#COPY nt_defense
D   Strength, ca 15+
2C  At least 5-4 majors
  (D)
    P  5+!c, suggestion to play
    R  Asking for better/longer major
    2D 5+!d, suggestion to play
  (P)
    2D Asking for better/longer major
2D  A weak major or a strong minor
  (P)
    2H Pass/correct
    2S Pass/correct
    2N Asking
2HS Constructive
2N  5-5 minors
3X  Preemptive
#ENDCOPY

  Note that the above is only for a direct overcall over 1NT. To
  define the above also when balancing we will use BML's
  copy/cut/paste functionality in order to show that you do not
  have to write it all over again. Take a look below (only visible in
  the /example.txt/ or generated Full Disclosure file, not in HTML or LaTeX):

#HIDE
(1N)-P-(P)---
#PASTE nt_defense
