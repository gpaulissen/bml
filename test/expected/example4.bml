#VUL 00

#SEAT 0

(1N)
  D = Strength, ca 15+
  2C = At least 5-4 majors
    (D)
      P = 5+!c, suggestion to play
      R = Asking for better/longer major
      2D = 5+!d, suggestion to play
    (P)
      2D = Asking for better/longer major
  2D = A weak major or a strong minor
    (P)
      2H = Pass/correct
      2S = Pass/correct
      2N = Asking
  2H = Constructive
  2S = Constructive
  2N = 5-5 minors
  3C = Preemptive
  3D = Preemptive
  3H = Preemptive
  3S = Preemptive

(1N)
  P
    (P)
      D = Strength, ca 15+
      2C = At least 5-4 majors
        (D)
          P = 5+!c, suggestion to play
          R = Asking for better/longer major
          2D = 5+!d, suggestion to play
        (P)
          2D = Asking for better/longer major
      2D = A weak major or a strong minor
        (P)
          2H = Pass/correct
          2S = Pass/correct
          2N = Asking
      2H = Constructive
      2S = Constructive
      2N = 5-5 minors
      3C = Preemptive
      3D = Preemptive
      3H = Preemptive
      3S = Preemptive
