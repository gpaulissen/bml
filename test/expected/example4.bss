*00{}=NYYYYYY
*001N=NYYYYYY0
*001ND=NYYYYYY0Strength, ca 15+
*001N2C=NYYYYYY008At least 5-4 majors
*001N2CD=NYYYYYY0
*001N2CDP=NYYYYYY05+!c, suggestion to play
*001N2CDR=NYYYYYY0Asking for better/longer major
*001N2CD2D=NYYYYYY0085+!d, suggestion to play
*001N2CP=NYYYYYY0
*001N2CP2D=NYYYYYY008Asking for better/longer major
*001N2D=NYYYYYY008A weak major or a strong minor
*001N2DP=NYYYYYY0
*001N2DP2H=NYYYYYY008Pass/correct
*001N2DP2S=NYYYYYY008Pass/correct
*001N2DP2N=NYYYYYY0Asking
*001N2H=NYYYYYY008Constructive
*001N2S=NYYYYYY008Constructive
*001N2N=NYYYYYY05-5 minors
*001N3C=NYYYYYY008Preemptive
*001N3D=NYYYYYY008Preemptive
*001N3H=NYYYYYY008Preemptive
*001N3S=NYYYYYY008Preemptive
*001NP=NYYYYYY0
*001NPP=NYYYYYY0
*001NPPD=NYYYYYY0Strength, ca 15+
*001NPP2C=NYYYYYY008At least 5-4 majors
*001NPP2CD=NYYYYYY0
*001NPP2CDP=NYYYYYY05+!c, suggestion to play
*001NPP2CDR=NYYYYYY0Asking for better/longer major
*001NPP2CD2D=NYYYYYY0085+!d, suggestion to play
*001NPP2CP=NYYYYYY0
*001NPP2CP2D=NYYYYYY008Asking for better/longer major
*001NPP2D=NYYYYYY008A weak major or a strong minor
*001NPP2DP=NYYYYYY0
*001NPP2DP2H=NYYYYYY008Pass/correct
*001NPP2DP2S=NYYYYYY008Pass/correct
*001NPP2DP2N=NYYYYYY0Asking
*001NPP2H=NYYYYYY008Constructive
*001NPP2S=NYYYYYY008Constructive
*001NPP2N=NYYYYYY05-5 minors
*001NPP3C=NYYYYYY008Preemptive
*001NPP3D=NYYYYYY008Preemptive
*001NPP3H=NYYYYYY008Preemptive
*001NPP3S=NYYYYYY008Preemptive
