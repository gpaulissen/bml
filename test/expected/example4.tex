\documentclass[a4paper]{article}
\usepackage[margin=1in]{geometry}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{newcent}
\usepackage{helvet}
\usepackage{graphicx}

\usepackage[pdftex, pdfborder={0 0 0}]{hyperref}
\frenchspacing

\usepackage{txfonts} % For \varheartsuit and \vardiamondsuit
\usepackage[usenames,dvipsnames]{color} % dvipsnames necessary to made PDFLaTeX work.
\usepackage{enumerate}
\usepackage{listliketab}
\usepackage{latexsym} % \Box
\usepackage{pbox} % \Box
\usepackage{parskip} % line between paragraphs

% suits

%%% Colors

\newcommand{\BC}{\textcolor{OliveGreen}{$\clubsuit$}}
\newcommand{\BD}{\textcolor{RedOrange}{$\vardiamondsuit$}}
\newcommand{\BH}{\textcolor{Red}{$\varheartsuit${}}}
\newcommand{\BS}{\textcolor{Blue}{$\spadesuit${}}}

%suits for pdf-friendly titles
\newcommand{\pdfc}{\texorpdfstring{\BC{}}{C}}
\newcommand{\pdfd}{\texorpdfstring{\BD{}}{D}}
\newcommand{\pdfh}{\texorpdfstring{\BH{}}{H}}
\newcommand{\pdfs}{\texorpdfstring{\BS{}}{S}}

\newenvironment{bidtable}
{\begin{tabbing}

xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=\kill}
{\end{tabbing} }%

\newenvironment{bidding}%
{\begin{tabbing}
xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx \kill
}{\end{tabbing} }%end bidding


% writing hands
\newcommand{\cards}[1]{\textsf{#1}}
\newcommand{\spades}[1]{\BS\cards{#1}}
\newcommand{\hearts}[1]{\BH\cards{#1}}
\newcommand{\diamonds}[1]{\BD\cards{#1}}
\newcommand{\clubs}[1]{\BC\cards{#1}}
\newcommand{\void}{--}
\newcommand{\hand}[4]{\spades{#1}\ \hearts{#2}\ \diamonds{#3} \clubs{#4}}
\newcommand{\vhand}[4]{\spades{#1}\\\hearts{#2}\\\diamonds{#3}\\\clubs{#4}}

% The \Box should always appear the same distance from the left margin
\newcommand\onesuit[4]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
                & \cards{#2} \\%
      \cards{#1}& $\Box$    & \cards{#3} \\%
                & \cards{#4} %
    \end{tabular}
  \end{center}%
}

% A special command if the south hand is not shown to avoid whitespace
\newcommand\onesuitenw[3]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}%
                & \cards{#2} \\%
      \cards{#1}& $\Box$    & \cards{#3}%
    \end{tabular}%
  \end{center}%
}

\newcommand\dealdiagram[5]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{\small #5}& \pbox{20cm}{#2} \\%
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#3} \\%
                & \pbox{20cm}{#4} %
    \end{tabular}
  \end{center}%
}

\newcommand\dealdiagramenw[4]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{\small #4}& \pbox{20cm}{#2} \\%
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#3} \\%
    \end{tabular}
  \end{center}%
}

\newcommand\dealdiagramew[2]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#2} \\%
    \end{tabular}
  \end{center}%
}

\title{No title defined}
\author{No author defined}
\begin{document}
\maketitle
\tableofcontents

\begin{bidtable}
(1NT)--\+\\
Dbl \> Strength, ca 15+\\
2\BC \> At least 5-4 majors\+\\
(D)\+\\
Pass \> 5+\BC , suggestion to play\\
Rdbl \> Asking for better/longer major\\
2\BD \> 5+\BD , suggestion to play\-\\
(P)\+\\
2\BD \> Asking for better/longer major\-\-\\
2\BD \> A weak major or a strong minor\+\\
(P)\+\\
2\BH \> Pass/correct\\
2\BS \> Pass/correct\\
2NT \> Asking\-\-\\
2\BH\BS \> Constructive\\
2NT \> 5-5 minors\\
3X \> Preemptive\-
\end{bidtable}
\bigbreak
Note that the above is only for a direct overcall over 1NT. To
define the above also when balancing we will use BML's
copy/cut/paste functionality in order to show that you do not
have to write it all over again. Take a look below (only visible in
the \emph{example.txt} or generated Full Disclosure file, not in HTML or LaTeX):
\bigbreak
\end{document}
